fl33t/models/mixins.py

```diff
@@ -25,20 +27,51 @@ def builds(self, offset=None, limit=None):
 
 class OneBuildMixin:  # pylint: disable=too-few-public-methods
     """For models with a build parent"""
+
+    _build = None
+
+    @property
     def build(self):
         """Return the parent build"""
-        return self._client.get_build(self.build_id)
+        if not self.build_id:
```

Comment:
I feel like, for the sake of maintaining consistency, we should always check `self._build.build_id` instead of this pseudo-double-cache here.

Reply:
This is to prevent a useless call to the fl33t API. Take for example a case where you have retrieved (or created) a device that has no `build_id` set. In that case, there is no associated build to retrieve. So rather than attempt the fl33t API call with an empty `build_id` which would result in an exception of some sort, it returns `None` immediately.

It would be possible to use a `try`/`catch` block to handle the thrown exception from the API, but in terms of pure performance, I figured that preventing the call entirely would be the better option. The same reasoning was used for when there is no `train_id` available to the device (via `self.fleet.train_id`) or to the fleet (via `self.train_id`).

Thoughts?

Reply:
> Take for example a case where you have retrieved (or created) a device that has no build_id set.

I hadn't thought of that situation… that's an interesting one. Ok, so let's proceed with the current implementation, and see how that goes. If it actually is troublesome maybe it's worth the latency.

At some point in the future we could have a `fresh` flag on the instance to indicate whether or not the instance is created locally but not yet persisted remotely, and in the situation where the instance was fetched remotely but just doesn't have a `build_id` associated to it, we could set a sentinel value on the instance so we know to not go and hit the API unless we force it to (e.g. a `refresh()` type of operation).

Reply:
Agreed. I know that our use-cases limit the need for `refresh()` functionality currently, but that will probably be required at some point. So I'll leave this as is for the moment and we can deal with `refresh()` functionality on it's own.